src/GreenPlayer.js

```diff
@@ -77,22 +120,32 @@ class SoundPlayer extends EventEmitter {
         this.initialized = false;
     }
 
+    /**
+     * Take the internal state of this player and create a new player from
+     * that. Restore the state of this player to that before its first playback.
+     *
+     * The returned player can be used to stop the original playback or
+     * continue it without manipulation from the original player.
+     *
+     * @returns {SoundPlayer} - new SoundPlayer with old state
+     */
     take () {
         if (this.outputNode) {
-            this.outputNode.removeEventListener('end', this.onEnd);
+            this.outputNode.removeEventListener(ON_ENDED, this.onEnded);
```

Comment:
this is one of those situations where I wonder if using the `object listener` version of add/remove event listener would be better...

```js
this.outputNode.addEventListener(ON_ENDED, this);

// and define on the class
handleEvent(event) {
  if (event.type === 'ON_ENDED') { this.onEnded(); }
}
```

It has a pretty nice effect of not needing to do a bound method in constructor.

Also, is there any reason to add/remove the listener over keeping it attached and toggling if we are waiting to end or not internally?

Reply:
It's removed during take because take is similar to stop. If the sound is playing taking the internal state is like stopping this sound. So when the original outputNode that persists on the SoundPlayer clone actually stops the original doesn't need to know it already thinks the sound stopped.

Reply:
I'm going to do the `object listener` change. 👍

Reply:
can you also add a comment with the reason to remove on take? thanks